### PR TITLE
sink(ticdc): precheck the sink URI and give some hints

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/6152

### What is changed and how it works?

- add precheck for sinkURI
- when we use wrong IPv6 format, give some hints

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
Manual testing:
```log
1. ./cdc cli changefeed create --pd="http://127.0.0.1:2379" --sink-uri="mysql://root@::1]:3306/" --changefeed-id="simple-replication-task1" --sort-engine="unifaied"
2. [CDC:ErrSinkURIInvalid]sink uri host is not valid IPv6 address, when using IPv6 address in URI, please use [ipv6-address]:port
```

File: cdc/sink/validator.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// Maybe we should support the dry-run mode to validate sink.
+func Validate(ctx context.Context, sinkURI string, cfg *config.ReplicaConfig) error {
+	if err := preCheckSinkURI(sinkURI); err != nil {
+		return err
+	}
+
+	errCh := make(chan error)
+	ctx, cancel := context.WithCancel(contextutil.PutRoleInCtx(ctx, util.RoleClient))
+	s, err := New(ctx, model.DefaultChangeFeedID("sink-verify"),
+		sinkURI, cfg, errCh)
```
Comment (overvenus) [inline cdc/sink/validator.go:38]:
```suggestion
	s, err := New(ctx, model.DefaultChangeFeedID("sink-verify"), sinkURI, cfg, errCh)
```

File: pkg/util/uri.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import "strings"
+
+// IsValidIPv6AddressInURI reports whether hostPort is a valid IPv6 address in URI.
+// See: https://www.ietf.org/rfc/rfc2732.txt.
+func IsValidIPv6AddressInURI(hostPort string) bool {
```
Comment (overvenus) [inline pkg/util/uri.go:20]: What about [`net.ParseIP`](https://pkg.go.dev/net#ParseIP)?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: pkg/util/uri.go (hunk 2 above)
Comment (hi-rustin) [reply pkg/util/uri.go:20]: The main purpose of this function is to determine the format in the URI, not whether it is an IP.

File: cdc/sink/validator.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sink/validator.go]: Done. Thanks!

File: pkg/util/uri.go (hunk 2 above)
Comment (hi-rustin) [reply pkg/util/uri.go:23]: Added in IsIPv6Address.

**Review by overvenus — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**